### PR TITLE
Workaround for checkbox cli

### PR DESCRIPTION
**Pull request by kiya956**
impact:
    sanity/agent/checkbox.py

description:
    current checkbox remote returncode is always 1 not matter success or fail
    So our workaround is not using returncode but check report file

test:
    N/A

### CODE REVIEW

**Comment by tsunghanliu** [comment]
Is there a checkbox bug filed about the return code?
can we keep the check of the return code but also check the report as fallback? In case the checkbox issue is fixed in the future.

**Comment by kiya956** [comment]
> Is there a checkbox bug filed about the return code? can we keep the check of the return code but also check the report as fallback? In case the checkbox issue is fixed in the future.

I prefer to revert this patch in the future, Stanley is working on the checkbox return issue.

File: sanity/agent/checkbox.py [hunk 1]
```diff
@@ -27,28 +28,28 @@ def run_checkbox(con, runner_cfg, secure_id, desc):
...
+        report = "failed to submit report"
         if status == 0:
             try:
                 report = re.search(
                     r"(?P<url>https?://certification.canonical.com[^\s]+)",
                     result,
                 ).group("url")
             except AttributeError:
-                report = "failed to submit report"
+                print("report url is not found")
```
Comment (stanley31huang) [inline sanity/agent/checkbox.py:53]:
```suggestion
                print("report url is not found")
        else:
            report = "failed to submit report"
```

File: sanity/agent/checkbox.py [hunk 2]
```diff
@@ -27,28 +28,28 @@ def run_checkbox(con, runner_cfg, secure_id, desc):
...
+    # if status == 0:
+    if os.path.exists("report.tar.xz"):
         upload_command = (
             f'checkbox.checkbox-cli submit -m "{desc}" '
             f"{secure_id} report.tar.xz"
         )
         print(upload_command)
 
         status, result = syscmd(upload_command)
+        report = "failed to submit report"
```
Comment (stanley31huang) [inline sanity/agent/checkbox.py:44]:
```suggestion
```

Comment (stanley31huang) [reply sanity/agent/checkbox.py:44]: I would suggest we move this line to following suggestion. It's more straightforward.

**Comment by stanley31huang** [comment]
> > Is there a checkbox bug filed about the return code? can we keep the check of the return code but also check the report as fallback? In case the checkbox issue is fixed in the future.
> 
> I prefer to revert this patch in the future, Stanley is working on the checkbox return issue.

I am checking with Cert team, I will send a patch after I got any feedback from Cert team.

**Review by stanley31huang — APPROVED** [review]
LGTM

**Review by tsunghanliu — APPROVED** [review]
LGTM

**Comment by stanley31huang** [comment]
@kiya956 @tsunghanliu 

the checkbox will return exit code 1 if any test cases failed. So that's the reason why we always get the exit code 1 (we did not get the all passed results).

**Pull request merged**